scripts/devenv.py

```diff
@@ -1,272 +1,109 @@
+
+    ap_sub = ap.add_subparsers(required=True, dest="cmd", metavar="cmd")
+
+    ap_create = ap_sub.add_parser(
+        "create",
+        help="create a new environment",
+        usage="%(prog)s [-h] [arguments ...]",
+        epilog=(
+            "All arguments will be added to the 'conda create' command.\n\n"
+            "default arguments:\n  " + " ".join(DEFAULT_CREATE_ARGS)
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
     )
+    ap_create.set_defaults(func=cmd_create)
 
-    remove_ap = sub.add_parser(
-        "remove",
-        help="remove an existing development environment",
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-    )
-    remove_ap.set_defaults(func=command_remove)
-    add_global_args(remove_ap)
+    ap_remove = ap_sub.add_parser("remove", help="remove an existing environment")
+    ap_remove.set_defaults(func=cmd_remove)
 
-    args = ap.parse_args()
+    args, rest = ap.parse_known_args()
 
-    if shutil.which("conda") is None:
-        logging.error("conda is not installed")
-        return
+    if not all(map(shutil.which, EXECUTABLES)):
+        sys.exit("the following executables should exist in the system path: " + str(EXECUTABLES)[1:-1])
 
-    if run_returncode("conda", "build", "--version") != 0:
-        logging.error("conda-build is not installed; run 'conda install -n base -c conda-forge conda-build' to install")
-        return
+    if not all(map(is_git_repo, REPOSITORIES)):
+        sys.exit("the following directories should be valid git repositories: " + str(REPOSITORIES)[1:-1])
 
-    env = CondaEnv(args.name)
-    execution_time, _res = timed(args.func, args, env)
-    logging.info(f"finished {args.command} ({execution_time:.3f} seconds)")
+    logging.info("command %r started", args.cmd)
+    time_start = time.perf_counter()
 
+    try:
+        args.func(args, rest)
+    except subprocess.CalledProcessError:
+        logging.error("command %r failed", args.cmd)
+    except KeyboardInterrupt:
+        logging.error("command %r interrupted", args.cmd)
+    else:
+        logging.info("command %r completed", args.cmd)
 
-def command_create(args: argparse.Namespace, env: CondaEnv) -> None:
-    if env.exists():
-        logging.error(f"environment {env.name} already exists")
-        return
+    time_end = time.perf_counter()
+    logging.info("elapsed time: %.3f seconds", time_end - time_start)
 
-    location = args.location
-    if location is not None:
-        location = pathlib.Path(location).resolve()
-        if not location.is_dir():
-            logging.error(f"{location} is not a valid directory")
-            return
 
-    logging.info(
-        f"create environment {env.name}"
-        f" with packages [{', '.join(args.package)}]"
-        f" from channels [{', '.join(args.channel)}]"
-    )
-    env.create(packages=args.package, channels=args.channel)
+def cmd_create(args, rest):
```

Comment:
just so I understand correctly, the syntax will slightly change (I'm fine with that if it is the case):

Before:

```
python devev.py create -n idev -c all my channels here -p all my packackages here
```

new:

```
python devenv.py create -n idev -c all -c my -c channels  here all my  packages  here
```
is this how it's supposed to be? I actually like it, because it's the same as in conda.

Reply:
Yes, it just passes everything to `conda create`.